addons/mail/static/src/core/common/record.js

```diff
@@ -215,8 +215,8 @@ export class RecordList extends Array {
     }
     /** @param {R[]} records */
     unshift(...records) {
-        for (const val of records) {
-            const r = this._insert(val, (r3) => {
+        for (let i = records.length - 1; i >= 0; i--) {
```

Comment:
maybe `[...records].reverse()` ?

Reply:
I don't think we should do 2 array traversal per unshift, this looks costy and this is a very low level method!

Reply:
yeah I understand what you meant. What I am worried about is when fetching data from the memory, if the data is continuous in memory, the data in the for loop can be properly prefetched from memory to CPU (which is the most costly time for most of the calculations). But if the data is read randomly (in this case, the index is constantly decreasing in an inversed way), the time it takes to read the data may be longer because the CPU's prefetch will fail. `...` and `reserve` are built-in implementation methods of js, so even if we do three times array operations, it may still be faster than random reading when dealing with big database. (Of course, different language interpreters may handle it differently and may need to be tested for specific case. This part of the inference mainly comes from the previous experience of writing cpp lol). But it is ok to keep like this if we don't perf it. I am just saying "fast copy three times and continuous in memory" can be faster than "no copy but without prefetch" sometimes

Reply:
Don't know whether `reverse()` is faster than `i--` in practice. Both are relatively similarly idiomatic, so I would just pick one arbitrarily and move on ;)